### ISSUE TITLE
Deprecate watchdog()

### ISSUE THREAD

**Issue by ghost** [backdrop/backdrop-issues#4749]
I recently was confused by:

- The UI page is called 'Recent log messages'
- The module is called 'Database Logging' (`dblog`)
- The API function for adding messages is called `watchdog()`

My research took me to [this page](https://www.drupal.org/documentation/modules/watchdog) which explains that the term 'watchdog' comes from the name of the predecessor to our `dblog` module from Drupal 4 & 5!

Since we more-or-less use the terms 'log/logging' now (in both the UI and the module name), I propose deprecating the `watchdog*()` functions and creating more generic and consistent `log*()` functions instead.

**Comment by findlabnet** [comment]
Pay attention to backward compatibility (wrapper or whatever) because many contribs use this function.

**Comment by indigoxela** [comment]
Hm. Does this also mean that the database table should be renamed? - It's also named "watchdog".
And there are constants and image names and...

That's *a lot* of effort that brings no benefit for the usual user/admin - just to raise a personal concern. :wink:

**Comment by klonos** [comment]
I share the same concerns as @indigoxela

**Comment by alanmels** [comment]
Good catch! I remember having similar thoughts, but never cared to create an actual issue.
